nemo/collections/tts/models/univnet.py

```diff
@@ -0,0 +1,373 @@
+            return x
+
+        # create bias tensor
+        if self.stft_bias is None or self.stft_bias.shape[0] != audio.shape[0]:
+            audio_bias = self(spec=torch.zeros_like(mel, device=mel.device))
+            self.stft_bias, _ = stft(audio_bias)
+            self.stft_bias = self.stft_bias[:, :, 0][:, :, None]
+
+        audio_mags, audio_phase = stft(audio)
+        audio_mags = audio_mags - self.cfg.get("denoise_strength", 0.0025) * self.stft_bias
+        audio_mags = torch.clamp(audio_mags, 0.0)
+        audio_denoised = istft(audio_mags, audio_phase).unsqueeze(1)
+
+        return audio_denoised
+
+    def __setup_dataloader_from_config(self, cfg, shuffle_should_be: bool = True, name: str = "train"):
+        if "dataset" not in cfg or not isinstance(cfg.dataset, DictConfig):
+            raise ValueError(f"No dataset for {name}")
+        if "dataloader_params" not in cfg or not isinstance(cfg.dataloader_params, DictConfig):
+            raise ValueError(f"No dataloder_params for {name}")
+        if shuffle_should_be:
+            if 'shuffle' not in cfg.dataloader_params:
+                logging.warning(
+                    f"Shuffle should be set to True for {self}'s {name} dataloader but was not found in its "
+                    "config. Manually setting to True"
+                )
+                with open_dict(cfg["dataloader_params"]):
+                    cfg.dataloader_params.shuffle = True
+            elif not cfg.dataloader_params.shuffle:
+                logging.error(f"The {name} dataloader for {self} has shuffle set to False!!!")
+        elif not shuffle_should_be and cfg.dataloader_params.shuffle:
+            logging.error(f"The {name} dataloader for {self} has shuffle set to True!!!")
+
+        dataset = instantiate(cfg.dataset)
+        return torch.utils.data.DataLoader(dataset, collate_fn=dataset.collate_fn, **cfg.dataloader_params)
+
+    def setup_training_data(self, cfg):
+        self._train_dl = self.__setup_dataloader_from_config(cfg)
+
+    def setup_validation_data(self, cfg):
+        self._validation_dl = self.__setup_dataloader_from_config(cfg, shuffle_should_be=False, name="validation")
+
+    @classmethod
+    def list_available_models(cls) -> 'Optional[Dict[str, str]]':
+        list_of_models = []
+        model = PretrainedModelInfo(
+            pretrained_model_name="tts_univnet",
+            location="https://api.ngc.nvidia.com/v2/models/nvidia/nemo/tts_univnet/versions/1.0.0rc1/files/tts_univnet.nemo",
+            description="This model is trained on LJSpeech audio sampled at 22050Hz and mel spectrograms generated from Tacotron2, TalkNet, and FastPitch. This model has been tested on generating female English voices with an American accent.",
+            class_=cls,
+        )
+        list_of_models.append(model)
+        return list_of_models
+
+    # def load_state_dict(self, state_dict, strict=True):
```

Comment:
I suppose you can delete it

Reply:
Deleted with [79d8a91](https://github.com/NVIDIA/NeMo/pull/3186/commits/79d8a9120c1c44f21bbcf144f612b588c7db98f8)